src/components/pages/products-list/ProductsGrid/ProductsGrid.js

```diff
@@ -0,0 +1,82 @@
+/* eslint-disable max-len */
+import { productsData } from "../../../../mocks/products";
+import { categoriesData } from "../../../../mocks/product-categories";
+import { Styled } from "./Styled";
+
+import Card from "../../../common/Card";
+import Image from "../../../common/Image";
+import Title from "../../../common/Title";
+
+export function ProductsGrid({ activeFilters }) {
+  let activeList = [];
+  for (let filter in activeFilters) {
+    console.log(filter);
+    if (activeFilters[filter]) {
+      activeList.push(filter);
+    }
+  }
+
+  const categories =
+    (Object.keys(activeFilters).length === 0 &&
+    activeFilters.constructor === Object) || activeList.length === 0
+      ? categoriesData.results.map((category) => category.id)
+      : activeList;
+
+  const allProducts = productsData.results;
+
+  const products = () => {
+    let result = [];
+
+    categories.map((filter) => {
+      return allProducts
+        .filter((product) => product.data.category.id === filter)
+        .forEach((product) => result.push(product));
+    });
+
+    return result;
+  };
```

Comment:
The products can be stored in the state of the component and the update can be done when the filter object changes, using `useState` and `useEffect`:

```jsx
  const [products, setProducts] = useState(productsData.results);

  useEffect(() => {
    const allProducts = productsData.results;
    const activeList = Object.entries(activeFilters)
      .filter(([filter, isChecked]) => isChecked)
      .map(([filter]) => filter);
    const categories = activeList.length === 0
      ? categoriesData.results.map(({ id }) => id)
      : activeList;

    const filteredProducts = allProducts.filter(product => (
      categories.includes(product.data.category.id)
    ));

    setProducts(filteredProducts);
  }, [productsData, activeFilters]);
```